src/app/cluster/details/external-cluster/update-external-cluster-machine-deployment-dialog/component.ts

```diff
@@ -0,0 +1,134 @@
+} from '@shared/entity/external-machine-deployment';
+import {BaseFormValidator} from '@shared/validators/base-form.validator';
+import {Observable} from 'rxjs';
+import {take, takeUntil} from 'rxjs/operators';
+import {MasterVersion} from '@shared/entity/cluster';
+
+class UpdateExternalClusterMachineDeploymentDialogData {
+  projectID: string;
+  clusterID: string;
+  machineDeployment: ExternalMachineDeployment;
+}
+
+enum Controls {
+  Replicas = 'replicas',
+  KubeletVersion = 'version',
+}
+
+@Component({
+  selector: 'km-update-external-cluster-machine-deployment-cluster-dialog',
+  templateUrl: './template.html',
+  styleUrls: ['./style.scss'],
+})
+export class UpdateExternalClusterMachineDeploymentDialogComponent extends BaseFormValidator implements OnInit {
+  readonly Controls = Controls;
+  disableReplicaControl: boolean;
+  versions: string[] = [];
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: UpdateExternalClusterMachineDeploymentDialogData,
+    private readonly _dialogRef: MatDialogRef<UpdateExternalClusterMachineDeploymentDialogComponent>,
+    private readonly _notificationService: NotificationService,
+    private readonly _externalMachineDeploymentService: ExternalMachineDeploymentService,
+    private readonly _builder: FormBuilder
+  ) {
+    super();
+  }
+
+  ngOnInit() {
+    this._initForm();
+    this._initSubscriptions();
+  }
+
+  getObservable(): Observable<any> {
+    const patchModel = this._updateExternalMachineDeploymentPatchModel();
+    const {projectID, clusterID, machineDeployment} = this.data;
+    return this._externalMachineDeploymentService
+      .patchExternalMachineDeployment(projectID, clusterID, machineDeployment.id, patchModel)
+      .pipe(take(1));
+  }
+
+  onNext(md: ExternalMachineDeployment): void {
+    this._notificationService.success(`Updated the ${md.name} machine deployment`);
+    this._dialogRef.close();
+  }
+
+  private _initForm(): void {
+    this.form = this._builder.group({
+      [Controls.Replicas]: this._builder.control(1),
+      [Controls.KubeletVersion]: this._builder.control(''),
+    });
+  }
+
+  private _initSubscriptions(): void {
+    this.form.valueChanges.pipe(takeUntil(this._unsubscribe)).subscribe(_ => {
+      this._updateExternalMachineDeploymentPatchModel();
+    });
+
+    const initialValue = this.form.value;
+    this.form
+      .get(Controls.Replicas)
+      .valueChanges.pipe(takeUntil(this._unsubscribe))
+      .subscribe(value => {
+        if (initialValue[Controls.Replicas] === value) {
+          this.form.get(Controls.KubeletVersion).enable({emitEvent: false});
+        } else {
+          this.form.get(Controls.KubeletVersion).disable({emitEvent: false});
+        }
+      });
+
+    this.form
+      .get(Controls.KubeletVersion)
+      .valueChanges.pipe(takeUntil(this._unsubscribe))
+      .subscribe(value => {
+        if (initialValue[Controls.KubeletVersion] === value) {
+          this.disableReplicaControl = false;
```

Comment:
Should we _also_ enable/disable `Replicas` form control similar to what we're doing for `KubeletVersion` above?

Reply:
IMO, we should kept as it is since Form provides method to disable form control. reason why we needed to have separate local instance variable.

- Use local variable to show dynamic tooltip
- km-stepper-number required `input` field

Reply:
I didn't ask to remove the local variable.

Reply:
Also conditions in `if` `else` should be swapped.

![screenshot-localhost_8000-2022 07 25-13_14_21](https://user-images.githubusercontent.com/13975988/180730659-06aa4008-6272-4b61-9f7f-fe7cd52f079c.png)

Reply:
Also, why isn't the kubelet version value pre-selected?

![screenshot-localhost_8000-2022 07 25-13_15_30](https://user-images.githubusercontent.com/13975988/180730878-31abebcf-315e-43be-97f7-0048eb84f2f7.png)

Reply:
Same comment for Replicas as well. It should show the existing value instead of hard-coded `1`